### PR TITLE
delete metadata entity sets and delete json from org

### PR DESCRIPTION
**Pull request by UnsungHero97**
A companion to https://github.com/openlattice/openlattice/pull/220. The main changes are:

- `DeleteOrgMetadataEntitySets` job that will delete legacy metadata entity sets and associated audit entity sets, and will also delete the `metadataEntitySetIds` stuff from the organization json
- to get `DeleteOrgMetadataEntitySets` working, I basically created a new test pod from scratch and only brought in the beans that were required to get it working. from there, I just copy/pasted what was in the test pod into `MechanicUpgradePod`
- deleted files in `/upgrades` related to metadata entity sets since they are no longer relevant

Note - the build will succeed after https://github.com/openlattice/openlattice/pull/220 is merged.

### CODE REVIEW

File: src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            auditEdgeEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+            auditRecordEntitySetIds.forEach { deleteEntitySet(orgId, it) }
+
+            if (deleteEntitySet(orgId, entitySetId)) {
+                return deleteMetadataFromJson(orgId, metadataKey)
+            }
+            // if the esid is all zeros AND there's no audit entity sets, it's likely the org wasn't initialized
+            // correctly. since there's no entity sets to delete, we'll try deleting metadata from json.
+            else if (
+                entitySetId.toString() == "00000000-0000-0000-0000-000000000000"
```
Comment (anzioka) [inline src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: UUID(0, 0) returns a UUID with all zeros

Comment (anzioka) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: Also, could one of `{ auditEdgeEntitySetIds, auditRecordEntitySetIds }` be empty and the other non-empty? That is, shouldn't this be `a && (b || c)`

File: src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            return true
+        } catch (e: Exception) {
+            logger.error("error deleting entity set - org $orgId entity set $entitySetId", e)
+        }
+
+        return false
+    }
+
+    private fun waitUntilJobFinishedOrCanceled(jobId: UUID, orgId: UUID, entitySetId: UUID) {
+
```
Comment (anzioka) [inline src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: Thread blocks until job status is one of { finished, paused, canceled } - the function name is confusing.

File: src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt (hunk 1 above)
Comment (UnsungHero97) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: I specifically want both to be empty. I only want to delete from json only if there's no entity sets left to delete. If an org is in a bad state, not initialized correctly, I want to leave the json alone so I can debug later and maybe manually do the cleanup. 

Whether or not this logic is "correct" or desirable is a different question, but it is what I intended.

File: src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt (hunk 2 above)
Comment (UnsungHero97) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: Fixed.

File: src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt (hunk 1 above)
Comment (UnsungHero97) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/DeleteOrgMetadataEntitySets.kt]: Updated with `UUID(0, 0)`.

**Review by geekbeast — APPROVED** [review]

**Review by anzioka — APPROVED** [review]

**Pull request merged**